Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/components/TimeStatsController.java

```diff
@@ -71,6 +86,41 @@ public void initializeControls(Object data) {
         logEndTime = (Label) compTimeStats.getFellow("endTime");
         logEndTime.setValue("-");
     }
+    
+    @Override
+    public void initializeEventListeners(Object data) throws Exception {
+        spnCaseHeading.addEventListener("onClick", new EventListener<Event>() {
+            @Override
+            public void onEvent(Event event) throws Exception {
+                if (disabled) return;
+                parent.openLogFilter(new Event("", null, "CaseTabPerformance"));
+            }
+        });
+        
+        lblCaseHeading.addEventListener("onClick", new EventListener<Event>() {
+            @Override
+            public void onEvent(Event event) throws Exception {
+                if (disabled) return;
+                parent.openLogFilter(new Event("", null, "CaseTabPerformance"));
+            }
+        });
+        
+        spnLogHeading.addEventListener("onClick", new EventListener<Event>() {
+            @Override
+            public void onEvent(Event event) throws Exception {
+                if (disabled) return;
+                parent.openLogFilter(new Event("", null, "CaseTabTimeframe"));
+            }
+        });
+        
+        lblLogHeading.addEventListener("onClick", new EventListener<Event>() {
+            @Override
+            public void onEvent(Event event) throws Exception {
+                if (disabled) return;
+                parent.openLogFilter(new Event("", null, "CaseTabTimeframe"));
+            }
+        });
```

Comment:
Similarly, all these event handler registrations can probably be abbreviated.  (I hope I'm right about that, otherwise this will be a lot of erroneous suggestions!)